chainer/testing/link.py

```diff
@@ -0,0 +1,549 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import backend
+from chainer import initializers
+from chainer.testing import array as array_module
+from chainer.testing import test
+
+
+class LinkTestError(test.TestError):
+    """Raised when the target link is implemented incorrectly."""
+    pass
+
+
+class InitializerPair(object):
+
+    """Class to hold a pair of initializer-like objects.
+
+    The initializer-like objects can be accessed via the ``first`` and the
+    ``second`` attributes.
+
+    When implementing ``LinkTestCase``, instances of this class can be included
+    in lists listing all initializer-like objects that should be tested in the
+    initializers test.
+    In that case, the first element should correspond to the initializer-like
+    argument passed to the constructor of the link, and the second element
+    correspond to the actual initializer-like object used by the link.
+    In many cases, they are the same, except when passing ``None`` to the
+    constructor.
+    """
+
+    def __init__(self, first, second):
+        initializers._check_is_initializer_like(first)
+        initializers._check_is_initializer_like(second)
+
+        self.first = first
```

Comment:
I think `first` and `second` are too general.
Isn't there any better naming?

Reply:
What do you think about `before_init`, `after_init`?

Reply:
How about `agrument_value` vs `expected_initializer`?

Reply:
Changes as suggested.